src/Model/DataModel/StreamData/HoldEmailData.php

```diff
@@ -53,6 +58,23 @@ public function assembleData(Patron $patron, Bib $bib, Item $item, HoldRequest $
         $this->setTitle($bib->getTitle());
         $this->setDeliveryLocation($holdRequest->getDeliveryLocation());
         $this->setPickupLocation($holdRequest->getPickupLocation());
+
+        /**
+         * @var DocDeliveryData
+         */
+        $docDeliveryData = $holdRequest->getDocDeliveryData();
+        $email = '';
+
+        if ($docDeliveryData !== null) {
+            $email = $docDeliveryData->getEmailAddress();
+        }
+
+        // If request is not an EDD, use e-mail from patron's info.
+        if ($email !== '') {
+            $this->setPatronEmail($email);
+        } else {
+            $this->setPatronEmail($patron->getEmails()[0]);
```

Comment:
is there a chance getEmails() will ever be null? If so, won't getEmails()[0] cause an error?